ykrt/src/mt.rs

```diff
@@ -255,69 +226,69 @@ impl MTThread {
         // to update values, allowing us to detect if we were interrupted. If we were interrupted,
         // we simply retry the whole operation.
         loop {
-            let pack = &loc.inner.pack;
             // We need Acquire ordering, as PHASE_COMPILED will need to read information written to
             // external data as a result of the PHASE_TRACING -> PHASE_COMPILED transition.
-            let lp = pack.load(Ordering::Acquire);
+            let lp = loc.pack.load(Ordering::Acquire);
             match lp & PHASE_TAG {
                 PHASE_COUNTING => {
                     let count = (lp & !PHASE_TAG) >> 2;
-                    let new_pack;
                     if count >= self.inner.hot_threshold {
                         if self.inner.tracer.is_some() {
                             // This thread is already tracing. Note that we don't increment the hot
                             // count further.
-                            break;
```

Comment:
This early return is functionally equivalent, right? Why the change?

Reply:
The loop is much more than a screen big, so I kept forgetting what `break` would jump to in this case.